packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js

```diff
@@ -163,10 +166,34 @@ export const useProductFullDetail = props => {
         productType
     );
 
-    const [{ isAddingItem }, { addItemToCart }] = useCartContext();
+    const [{ cartId, isAddingItem }, { addItemToCart }] = useCartContext();
+    console.log(`pfd cartId`, cartId);
 
+    const apolloClient = useApolloClient();
     const [addConfigurableProductToCart] = useMutation(
-        addConfigurableProductToCartMutation
+        addConfigurableProductToCartMutation,
+        {
+            onError: async err => {
+                console.log('got error', err);
+                // Refetch cart id when the mutation fails. I was hoping this
+                // could be done in CartContext but the lazy query is not being
+                // triggered. An alternative may be to do this in a link?
+                try {
+                    // We could specify to only re-create cart id for specific
+                    // errors.
+                    const { data } = await apolloClient.mutate({
+                        mutation: createCartMutation
+                    });
+
+                    apolloClient.writeQuery({
+                        query: GET_LOCAL_CART_ID,
+                        data
+                    });
```

Comment:
Initially I thought I could just write an empty object or `null` here to trigger the re-query in the cart context but that didn't work. It feels bad to have to handle recreation of the cart id in this way, especially when the retry pattern should be:

1) Validate/recreate auth token
2) Validate/recreate cart id